src/main/java/duke/misc/Parser.java

```diff
@@ -0,0 +1,271 @@
+package duke.misc;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidCommand;
+import duke.exception.InvalidDateTime;
+import duke.exception.InvalidIndex;
+
+/**
+ * Extracts and deciphers the user's input, and subsequently calls the relevant function.
+ * Type of inputs that are currently supported:
+ * list/todo/deadline/event/mark/unmark/delete/bye
+ *
+ * @author Terng Yan Long
+ */
+public class Parser {
+    /**
+     * Calls different functions according to the user's input command.
+     * Any command that is not in the correct form will be throw an "error" message.
+     *
+     * @param userInput Command entered by the user.
+     */
+    public static String parse(String userInput, TaskList taskListOfTasks) throws DukeException {
+        String[] wordsArray = userInput.trim().split(" ");
+        List<String> wordsList = Arrays.asList(wordsArray);
+        String firstWord = wordsList.get(0);
+
+        switch (firstWord) {

```

Comment:
Nice, I like this neatness of this code. I used an if-else because switch-case had issues I couldn't solve on my end but seems like the switch-case statements really makes it a lot neater looking.